python/mlcroissant/mlcroissant/_src/datasets.py

```diff
@@ -163,6 +168,35 @@ def __iter__(self):
                 record_set=self.record_set, operations=operations
             )
 
+    def beam_reader(self, pipeline: beam.Pipeline):
+        """Returns an Apache Beam sink to generate the dataset using Flume/Spark/etc.
+
+        Example of usage:
+
+        ```python
+        import apache_beam as beam
+        from apache_beam.options.pipeline_options import PipelineOptions
+        import mlcroissant as mlc
+
+        dataset = mlc.Dataset(
+            jsonld="https://huggingface.co/api/datasets/ylecun/mnist/croissant",
+        )
+        pipeline_options = PipelineOptions()
+        with beam.Pipeline(options=pipeline_options) as pipeline:
+            _ = dataset.records("mnist").beam_reader(pipeline)
+        ```
+        """
+        operations = self._filter_interesting_operations(self.filters)
+        execute_downloads(operations)
+        can_stream_dataset = all(d == 1 or d == 2 for _, d in operations.degree())
```

Comment:
NIT: I know you explain this below, but I would mention also in the docstring of this function 1) that it raises a ValueError for non-streamable datasets, and 2) what streamable datasets are.

Reply:
Done.